From 873103c8862da4e2d6df424da1eb1e4a344f28d6 Mon Sep 17 00:00:00 2001
From: Assam Boudjelthia <assam.boudjelthia@qt.io>
Date: Tue, 14 Oct 2025 17:37:10 +0300
Subject: [PATCH 402/553] CMake: fix CMP0177 warning under
 qt_internal_create_source_jar()

Fixes: QTBUG-135304
Change-Id: Ic65ac82d541091920ae100267fb543640c838e2e
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit 06c9cd7e4513066bc51e00939b2eca135fec36fd)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 cmake/QtAndroidHelpers.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmake/QtAndroidHelpers.cmake b/cmake/QtAndroidHelpers.cmake
index 64696fff6d4..4d7866d7bcf 100644
--- a/cmake/QtAndroidHelpers.cmake
+++ b/cmake/QtAndroidHelpers.cmake
@@ -455,8 +455,9 @@ function(qt_internal_create_source_jar)
     add_dependencies(android_source_jars ${jar_target})
 
     if(QT_WILL_INSTALL)
+        qt_path_join(destination "${INSTALL_DATADIR}" "android" "${module}")
         install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${jar_name}-${PROJECT_VERSION}.jar"
-            DESTINATION "${INSTALL_DATADIR}/android/${module}"
+            DESTINATION "${destination}"
             COMPONENT _install_android_source_jar_${module}
             EXCLUDE_FROM_ALL
         )
-- 
2.51.2

